stdlib/tests/crypto/rpo.rs

```diff
@@ -274,3 +274,96 @@ fn test_squeeze_digest() {
 
     build_test!(even_words, &[]).expect_stack(&even_hash);
 }
+
+#[test]
+fn test_hash_memory() {
+    // hash fewer than 8 elements
+    let compute_inputs_hash_5 = "
+    use.std::crypto::hashes::rpo
+
+    begin
+        push.1.2.3.4.1000 mem_storew dropw
+        push.5.0.0.0.1001 mem_storew dropw
+        push.11
+
+        push.5.1000
+
+        exec.rpo::hash_memory
+    end
+    ";
+
+    #[rustfmt::skip]
+    let mut expected_hash: Vec<u64> = build_expected_hash(&[
+        1, 2, 3, 4, 5
+    ]).into_iter().map(|e| e.as_int()).collect();
+    // make sure that value `11` stays unchanged
+    expected_hash.push(11);
+    build_test!(compute_inputs_hash_5, &[]).expect_stack(&expected_hash);
+
+    // hash exactly 8 elements
+    let compute_inputs_hash_8 = "
+    use.std::crypto::hashes::rpo
+
+    begin
+        push.1.2.3.4.1000 mem_storew dropw
+        push.5.6.7.8.1001 mem_storew dropw
+        push.11
+
+        push.8.1000
+
+        exec.rpo::hash_memory
+    end
+    ";
+
+    #[rustfmt::skip]
+    let mut expected_hash: Vec<u64> = build_expected_hash(&[
+        1, 2, 3, 4, 5, 6, 7, 8
+    ]).into_iter().map(|e| e.as_int()).collect();
+    // make sure that value `11` stays unchanged
+    expected_hash.push(11);
+    build_test!(compute_inputs_hash_8, &[]).expect_stack(&expected_hash);
+
+    // hash more than 8 elements
+    let compute_inputs_hash_15 = "
+    use.std::crypto::hashes::rpo
+
+    begin
+        push.1.2.3.4.1000 mem_storew dropw
+        push.5.6.7.8.1001 mem_storew dropw
+        push.9.10.11.12.1002 mem_storew dropw
+        push.13.14.15.0.1003 mem_storew dropw
+        push.11
+
+        push.15.1000
+
+        exec.rpo::hash_memory
+    end
+    ";
+
+    #[rustfmt::skip]
+    let mut expected_hash: Vec<u64> = build_expected_hash(&[
+        1, 2, 3, 4, 
+        5, 6, 7, 8, 
+        9, 10, 11, 12, 
+        13, 14, 15
+    ]).into_iter().map(|e| e.as_int()).collect();
+    // make sure that value `11` stays unchanged
+    expected_hash.push(11);
+    build_test!(compute_inputs_hash_15, &[]).expect_stack(&expected_hash);
+}
+
+#[test]
+fn test_hash_memory_empty() {
```

Comment:
Let's change this test to also test `hash_memory_words` and `absorb_double_words_from_memory`.

Reply:
Yes, I think we should. Is there a downside of doing this?

Reply:
No, at least as far as I can see — no, so I lifted this restriction.